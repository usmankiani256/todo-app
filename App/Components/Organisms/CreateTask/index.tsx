import * as React from 'react'
import { FAB } from 'react-native-paper'
import useStyles from './styles'
import { TaskDialog } from '@Molecules'

export type CreateTaskProps = {}

const CreateTask = (props: CreateTaskProps) => {
  const {} = props

  const [visible, setVisible] = React.useState(false)

  const showDialog = () => setVisible(true)

  const hideDialog = () => setVisible(false)

  const { root } = useStyles()

  return (
    <>
      <FAB
        testID="create-task-button"
        onPress={showDialog}
        style={root}
        icon="plus"
      />
      {visible && <TaskDialog hideDialog={hideDialog} />}
    </>
  )
}

export default CreateTask
