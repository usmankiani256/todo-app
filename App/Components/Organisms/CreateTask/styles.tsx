import { Dimensions, Platform, StyleSheet } from 'react-native'
import { Colors } from 'Theme'

const { width } = Dimensions.get('window')

const useStyles = () =>
  StyleSheet.create({
    root: {
      position: 'absolute',
      right: 20,
      bottom: 40,
      backgroundColor: Colors.primary,
    },
  })

export default useStyles
