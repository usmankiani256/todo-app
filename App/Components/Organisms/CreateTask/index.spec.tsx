import * as React from 'react'
import { render, fireEvent } from '@testing-library/react-native'
import CreateTask from './index'

jest.useFakeTimers()

describe('CreateTask Component', () => {
  it('renders correctly', () => {
    const rendered: any = render(<CreateTask />).toJSON()

    expect(rendered).toMatchSnapshot()
  })

  // it('opens dialog on click', () => {
  //   const rendered = render(<CreateTask />)
  //   const json = rendered.toJSON()

  //   fireEvent.press(rendered.getByTestId('create-task-button'))

  //   expect(json).toMatchSnapshot()
  // })
})
