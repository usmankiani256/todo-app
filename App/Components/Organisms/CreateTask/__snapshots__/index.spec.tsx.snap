// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateTask Component renders correctly 1`] = `
<View
  collapsable={false}
  nativeID="animatedComponent"
  pointerEvents="auto"
  style={
    Object {
      "backgroundColor": "#31394F",
      "borderRadius": 28,
      "bottom": 40,
      "elevation": 6,
      "opacity": 1,
      "position": "absolute",
      "right": 20,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 5,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 6,
      "transform": Array [
        Object {
          "scale": 1,
        },
      ],
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 28,
        },
      ]
    }
    testID="create-task-button"
  >
    <View
      pointerEvents="none"
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "height": 56,
            "width": 56,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "height": 24,
              "width": 24,
            },
          ]
        }
      >
        <View
          collapsable={false}
          nativeID="animatedComponent"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "rotate": "0deg",
                },
              ],
            }
          }
        >
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontSize": 24,
                },
                Array [
                  Object {
                    "lineHeight": 24,
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ],
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            Û∞êï
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
